src/SensitiveFloats.jl

```diff
@@ -271,8 +235,19 @@ function add_sources_sf!{ParamType <: ParamSet, NumType <: Number}(
             end
         end
     end
+end
 
-    true # Set definite return type
+
+function zero_sensitive_float_array(NumType::DataType,
+                                    local_P::Int,
+                                    local_S::Int,
+                                    d::Integer...)
+    sf_array = Array(SensitiveFloat{NumType}, d)
+    for ind in 1:length(sf_array)
+        # Do we always want these arrays to have gradients and hessians?
```

Comment:
Not yet, you may want to change this eventually. For the `SensitiveFloat` constructor, I prefer not having default values for `has_gradient` and `has_hessian` to make it explicit what memory is going to be allocated.